### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by afuetterer**
Hi, this is my first PR to the project. Please feel free to close or adapt. This PR fixes some minor typos.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by bgruening** [comment]
Welcome @afuetterer! That is a pretty good start to dive into the code base :)
Thanks!

File: test/unit/files/test_ftp.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def test_file_source_ftp_specific():
     assert_realizes_contains(
         file_sources,
         test_url,
-        "This is ftp.gnu.org, the FTP server of the the GNU project.",
+        "This is ftp.gnu.org, the FTP server of the GNU project.",
```
Comment (nsoranzo) [inline test/unit/files/test_ftp.py]: Can you please revert this change? The typo comes from the remote ftp.gnu.org/README

**Comment by afuetterer** [comment]
Thank you.

File: lib/galaxy/webapps/galaxy/controllers/dataset.py [hunk 2]
```diff
@@ -816,7 +816,7 @@ def _purge(self, trans, dataset_id):
                 try:
                     hda.dataset.full_delete()
                     trans.log_event(
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/dataset.py]:
```suggestion
                    trans.log_event(f"Dataset id {hda.dataset.id} has been purged upon the purge of HDA id {hda.id}")
```

File: lib/galaxy/webapps/galaxy/controllers/dataset.py [hunk 3]
```diff
@@ -816,7 +816,7 @@ def _purge(self, trans, dataset_id):
...
-                        f"Dataset id {hda.dataset.id} has been purged upon the the purge of HDA id {hda.id}"
+                        f"Dataset id {hda.dataset.id} has been purged upon the purge of HDA id {hda.id}"
                     )
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/dataset.py]:
```suggestion
```

File: lib/tool_shed/test/functional/test_0170_complex_prior_installation_required.py [hunk 4]
```diff
@@ -6,14 +6,14 @@
 )
 
 matplotlib_repository_name = "package_matplotlib_1_2_0170"
-matplotlib_repository_description = "Contains a tool dependency definition that downloads and compiles version 1.2.x of the the python matplotlib package."
+matplotlib_repository_description = "Contains a tool dependency definition that downloads and compiles version 1.2.x of the python matplotlib package."
```
Comment (nsoranzo) [inline lib/tool_shed/test/functional/test_0170_complex_prior_installation_required.py]:
```suggestion
matplotlib_repository_description = (
    "Contains a tool dependency definition that downloads and compiles version 1.2.x of the python matplotlib package."
)
```

File: lib/tool_shed/test/functional/test_1190_complex_prior_installation_required.py [hunk 5]
```diff
@@ -6,14 +6,14 @@
 )
 
 matplotlib_repository_name = "package_matplotlib_1_2_0170"
-matplotlib_repository_description = "Contains a tool dependency definition that downloads and compiles version 1.2.x of the the python matplotlib package."
+matplotlib_repository_description = "Contains a tool dependency definition that downloads and compiles version 1.2.x of the python matplotlib package."
```
Comment (nsoranzo) [inline lib/tool_shed/test/functional/test_1190_complex_prior_installation_required.py]:
```suggestion
matplotlib_repository_description = (
    "Contains a tool dependency definition that downloads and compiles version 1.2.x of the python matplotlib package."
)
```

**Review by nsoranzo — COMMENTED** [review]
Thanks for the fixes! I've suggested changes to make `black` happy.

File: test/unit/files/test_ftp.py (hunk 1 above)
Comment (afuetterer) [reply test/unit/files/test_ftp.py]: Sure thing.

File: test/unit/files/test_http.py [hunk 6]
```diff
@@ -87,7 +87,7 @@ def test_file_source_ftp_url():
 
     def check_generic_headers(request, **kwargs):
         assert not request.headers
-        response: Any = io.StringIO("This is ftp.gnu.org, the FTP server of the the GNU project.")
+        response: Any = io.StringIO("This is ftp.gnu.org, the FTP server of the GNU project.")
```
Comment (afuetterer) [inline test/unit/files/test_http.py]: Same here? Revert?

**Comment by jdavcs** [comment]
Very nice, thank you! I didn't realize we had so many the the cases :)

**Review by jdavcs — APPROVED** [review]

**Comment by afuetterer** [comment]
Happens very easily I guess. :)

File: test/unit/files/test_http.py (hunk 6 above)
Comment (afuetterer) [reply test/unit/files/test_http.py]: Alright, will revert and force push again.

**Comment by dannon** [comment]
What a weird typo to happen so frequently, thanks for the fixes!

**Comment by nsoranzo** [comment]
Thanks for the the fixes :laughing:
